filesystem/iso9660.ksy

```diff
@@ -5,269 +5,589 @@ meta:
+            'su_signature::rrip_symbolic_link': susp_unknown # SL
+            'su_signature::susp_indicator': susp_unknown # SP
+            'su_signature::susp_terminator': susp_unknown # ST
+            'su_signature::rrip_time_file': rrip_tf # TF
+            'su_signature::rrzf_zisofs': susp_unknown # ZF
+  su_headers:
+    seq:
+      - id: header
+        type: su_header
+        repeat: eos
+  directory_records:
+    doc: |
+      First item "." it points to it self
+      Second item ".." it points to the parent, or also to self if it the root
+    seq:
+      - id: directory_record
+        type: directory_record_head
+        repeat: until
+        repeat-until: _.len_dr == 0
+  directory_record_head:
+    doc-ref: ecma-119 9.1
+    seq:
+      - id: len_dr
+        doc-ref: ecma-119 9.1.1
         type: u1
-      - id: minute
+      - id: body
+        type: directory_record_body
+        if: len_dr > 0x0
+  directory_record_body:
+    seq:
+      - id: ext_attr_rec_len
+        doc-ref: ecma-119 9.1.2
         type: u1
-      - id: sec
+      - id: location_of_extent
+        doc-ref: ecma-119 9.1.3
+        type: u4bi
+      - id: data_len
+        doc-ref: ecma-119 9.1.4
+        type: u4bi
+      - id: rec_date_time
+        doc-ref: ecma-119 9.1.5
+        type: datetime_short
+      - id: file_flags_multi_extent
+        doc-ref: ecma-119 9.1.6 b7
+        type: b1
+      - id: file_flags_reserved
+        doc-ref: ecma-119 9.1.6 b5+b6
+        type: b2
+      - id: file_flags_protection
+        doc-ref: ecma-119 9.1.6 b4
+        type: b1
+      - id: file_flags_record
+        doc-ref: ecma-119 9.1.6 b3
+        type: b1
+      - id: file_flags_associated_file
+        doc-ref: ecma-119 9.1.6 b2
+        type: b1
+      - id: file_flags_directory
+        doc-ref: ecma-119 9.1.6 b1
+        type: b1
+      - id: file_flags_existence
+        doc-ref: ecma-119 9.1.6 b0
+        type: b1
+      - id: file_unit_size
+        doc-ref: ecma-119 9.1.7
         type: u1
-      - id: timezone
+      - id: interleave_gap_size
+        doc-ref: ecma-119 9.1.8
         type: u1
-  dec_datetime:
-    doc-ref: 'http://wiki.osdev.org/ISO_9660#Date.2Ftime_format'
-    seq:
-      - id: year
-        type: str
-        size: 4
-        encoding: ASCII
-      - id: month
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: day
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: hour
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: minute
-        type: str
-        size: 2
-        encoding: ASCII
-      - id: sec
+      - id: vol_seq_num
+        doc-ref: ecma-119 9.1.9
+        type: u2bi
+      - id: len_fi
+        doc-ref: ecma-119 9.1.10
+        type: u1
+      - id: file_id_file
+        doc-ref: ecma-119 9.1.11
         type: str
-        size: 2
-        encoding: ASCII
-      - id: sec_hundreds
+        size: len_fi
+        if: not file_flags_directory
```

Comment:
```switch-on```?

Reply:
Maybe I don't understand.
The switch-on would leave me a single variable containing:
- the content of the file
- more nested directory structures

Giving them different variable names actually helps here I think.

Reply:
>The switch-on would leave me a single variable containing
    * the content of the file
    * more nested directory structures

Yes, and IMHO it's good.